### ISSUE TITLE
Cannot send message to users with old last chat time  

### ISSUE THREAD

**Issue by RYNEQ** [vysheng/tg#1413]
Hi
I'm trying to send message to a user using 
`msg @username "text"`
When I use this command on users whom I had conversation with them recently  it works neatly. 
But when last conversation is old (or with no conversation) tg-cli answers:
`> FAIL: 38: can not parse arg #1`

What is the problem?
What is the correct way to send message to users with no recent conversation?

**Comment by wildshaman** [comment]
You must use `resolve_username username` before using msg @ or other methods with usernames.

**Comment by marcogaio** [comment]
I've just compiled telegram-cli on debian jessie.
I need it rarely in script, so i don't want to enable daemon mode.

But in script if i try to sent messages:
`asterisk@eraldo:~$ telegram-cli -DCR -e 'msg Marco_Gaiarin "Ciao."'
FAIL: 38: can not parse arg #1
All done. Exit
halt
`

But if i connect in cli mode (no -e ''), i send a message from Marco_Gaiarin, then i can send to:
`asterisk@eraldo:~$ telegram-cli
Telegram-cli version 1.4.1, Copyright (C) 2013-2015 Vitaly Valtman
Telegram-cli comes with ABSOLUTELY NO WARRANTY; for details type `show_license'.
This is free software, and you are welcome to redistribute it
under certain conditions; type `show_license' for details.
Telegram-cli uses libtgl version 2.1.0
Telegram-cli includes software developed by the OpenSSL Project
for use in the OpenSSL Toolkit. (http://www.openssl.org/)
I: config dir=[/var/lib/asterisk/.telegram-cli]
> msg Marco_Gaiarin ciao.
FAIL: 38: can not parse arg #1
 [18:45]  Marco Gaiarin »»» Ciao.
> msg Marco_Gaiarin ciao.
 [18:45]  Marco Gaiarin <<< ciao.
User Marco Gaiarin marked read 1 outbox and 0 inbox messages
> quit
halt
`

I've tried resolve_username, but:
`asterisk@eraldo:~$ telegram-cli -DCR -e 'resolve_username Marco_Gaiarin'
FAIL: 71: RPC_CALL_FAIL 400: USERNAME_NOT_OCCUPIED
All done. Exit
halt
`

Someone have some hints? Thanks.

**Comment by Paulchen232** [comment]
I think Marco_Gaiarin ist not the username :-). Use the username with resolve_username without @. If you don't now what the username is look the Telegram FAQ.

**Comment by Paulchen232** [comment]
And use with `msg', if you don't have add the contact or a chat with the target, the username.

**Comment by marcogaio** [comment]
`asterisk@eraldo:~$ telegram-cli -DCR -e 'resolve_username marcogaio'
Marco Gaiarin
All done. Exit
halt
`
ok, this seems to work. But still no message get sent:

`asterisk@eraldo:~$ telegram-cli -DCR -e 'msg @marcogaio "Ciao."'
FAIL: 38: can not parse arg #1
All done. Exit
halt
asterisk@eraldo:~$ telegram-cli -DCR -e 'msg "@marcogaio" "Ciao."'
All done. Exit
halt
asterisk@eraldo:~$ telegram-cli -DCR -e 'msg "Marco Gaiarin" "Ciao."'
All done. Exit
halt
asterisk@eraldo:~$ telegram-cli -DCR -e 'msg marcogaio "Ciao."'
FAIL: 38: can not parse arg #1
All done. Exit
halt
asterisk@eraldo:~$ telegram-cli -DCR -e 'msg "marcogaio" "Ciao."'
All done. Exit
halt
`

And also withouth error, NO messages arrived. But if i try in CLI:
`> resolve_username marcogaio
Marco Gaiarin
> msg @marcogaio Ciao.
 [12:07]  Marco Gaiarin <<< Ciao.
`

So seems i've to send 'resolve_username' and 'msg' in the same ''session'': but how?! I've trie to use multiple '-e' command or pipe commands to telegram-cli without success.

Thanks.

**Comment by Paulchen232** [comment]
You get no answer because you use the -D Argument. You have an chat so it works if you use the -W Argument.

**Comment by marcogaio** [comment]
No, with 'NO messages arrived' i mean that no message arrived at @marcogaio telegram account, mine. ;-)

After writing here i've make some other esperiment, and seems that work if i use multiple '-e' commands. But strangely, only the first and the last seems to work (and adding '-W' option, thanks):

`
asterisk@eraldo:~$ /usr/bin/telegram-cli -WDCR -e "resolve_username marcogaio" -e "msg @marcogaio Test1." -e "msg @marcogaio Test2."
All done. Exit
halt
`

eg, only 'test2' message arrived.

**Comment by Paulchen232** [comment]
It work only the last -e Argument.

**Comment by marcogaio** [comment]
OK, i can confirm: to simply send a message '-CDWR -e ""' seems the right commandline.

But... if i need to send to a chat?

**Comment by Paulchen232** [comment]
You mean you want to write to a chat name you never wrote before? Or what you mean?

I think now with -W and username it works.

**Comment by Paulchen232** [comment]
You don`t need it now but you can Multiple '-e' commands use with ' and a real line break. So:
pi@raspberrypi:~ $ ~/tg/bin/telegram-cli -W
 -e 'msg @XyZ hallo7
msg @XyZ hallo8'
